### PR TITLE
Play 2.3 support

### PR DESCRIPTION
**Pull request by devinsba**
Adds instrumentation for Play 2.3. This is mostly a copy+paste from Play 2.4 with changes where needed

### CODE REVIEW

File: dd-java-agent/instrumentation/play-2.3/src/test/scala/server/SyncServer.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      HttpServerTest.controller(SUCCESS, new ControllerClosureAdapter(Results.Status(SUCCESS.getStatus).apply(SUCCESS.getBody)))
+    }
+    case ("GET", "/redirect") => Action { request =>
+      HttpServerTest.controller(REDIRECT, new ControllerClosureAdapter(Results.Redirect(REDIRECT.getBody, REDIRECT.getStatus)))
+    }
+    case ("GET", "/query") => Action { request =>
+      HttpServerTest.controller(QUERY_PARAM, new ControllerClosureAdapter(Results.Status(QUERY_PARAM.getStatus).apply(QUERY_PARAM.getBody)))
+    }
+    case ("GET", "/error-status") => Action { request =>
+      HttpServerTest.controller(QUERY_PARAM, new ControllerClosureAdapter(Results.Status(ERROR.getStatus).apply(ERROR.getBody)))
```
Comment (devinsba) [inline dd-java-agent/instrumentation/play-2.3/src/test/scala/server/SyncServer.scala]: Hmmm. This test shouldn’t be passing 

```suggestion
      HttpServerTest.controller(ERROR, new ControllerClosureAdapter(Results.Status(ERROR.getStatus).apply(ERROR.getBody)))
```

Comment (devinsba) [reply dd-java-agent/instrumentation/play-2.3/src/test/scala/server/SyncServer.scala]: eh, it only matters for `NOT_FOUND`

**Comment by randomanderson** [comment]
If #1320 is merged first, there might be more work to do here with tests

**Comment by devinsba** [comment]
Might as well merge that first since I'm around

**Pull request merged**